src/shogun/machine/LinearMachine.cpp

```diff
@@ -134,3 +134,52 @@ CDotFeatures* CLinearMachine::get_features()
 void CLinearMachine::store_model_features()
 {
 }
+
+// TODO: almost the same as CMachine::train, move this method to CMachine after
+// we have done refactoring other classes
+void CLinearMachine::fit(CFeatures* features, CLabels* labels)
+{
+	REQUIRE(train_require_labels(), "Labels are not needed.");
+	labels->ensure_valid(get_name());
+	/* not allowed to train on locked data */
+	REQUIRE(
+	    !m_data_locked, "(%s)::train data_lock() was called, only "
+	                    "train_locked() is possible. Call data_unlock if you "
+	                    "want to call train()\n",
+	    get_name());
+
+	auto sub = connect_to_signal_handler();
+
+	if (support_feature_dispatching())
```

Comment:
@shubham808 maybe have a read of those changes since you wrote the original code

Reply:
this seems same as CMachine::train so it will not cause any problems but i see it is planned to be moved there after refactor